app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,66 @@
+class PaymentsController < ApplicationController
+
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  def index
+    @payments = current_company.payments
+  end
+
+  def new
+    @payment = Payment.new
+    find_invoice(params[:id])
+  end
+
+  def edit
+    find_invoice(@payment.invoice_id)
+    params[:id] = @payment.invoice_id
+  end
+
+  def create
+    @payment = Payment.new(payment_params)
+    find_invoice(params[:id])
+    if @payment.save
+      if @payment.invoice.total_amount_payments < @payment.invoice.total_amount
```

Comment:
Status change should be done in the `Payment` model itself in `after_save/after_create` callback